### PR TITLE
feat: add role connections

### PR DESCRIPTION
**Pull request by almeidx**
**Please describe the changes this PR makes and why it should be merged:**

**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**
- discord/discord-api-docs#5668

### CODE REVIEW

File: deno/payloads/v10/user.ts [hunk 1]
```diff
@@ -255,3 +255,18 @@ export enum ConnectionVisibility {
 	 */
 	Everyone,
 }
+
+export interface ApplicationRoleConnection {
```
Comment (vladfrangu) [inline deno/payloads/v10/user.ts:259]: Ditto

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/payloads/v10/application.ts [hunk 2]
```diff
@@ -131,3 +137,71 @@ export enum ApplicationFlags {
...
+ */
+export enum ApplicationRoleConnectionMetadataType {
+	/**
+	 * The metadata value (`integer`) is less than or equal to the guild's configured value (`integer`)
+	 */
+	IntegerLessThanOrEqual = 1,
+	/**
+	 * The metadata value (`integer`) is greater than or equal to the guild's configured value (`integer`)
+	 */
+	IntegerGreatThanOrEqual,
```
Comment (vladfrangu) [inline deno/payloads/v10/application.ts:182]: I'd call this `GreaterThan` not `GreatThan`

File: deno/payloads/v10/application.ts [hunk 3]
```diff
@@ -131,3 +137,71 @@ export enum ApplicationFlags {
...
+	 */
+	IntegerNotEqual,
+	/**
+	 * The metadata value (`ISO8601 string`) is less than or equal to the guild's configured value (`integer`; days before current date)
+	 */
+	DatetimeLessThanOrEqual,
+	/**
+	 * The metadata value (`ISO8601 string`) is greater than or equal to the guild's configured value (`integer`; days before current date)
+	 */
+	DatetimeGreatThanOrEqual,
```
Comment (vladfrangu) [inline deno/payloads/v10/application.ts:198]: Ditto

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/payloads/v10/application.ts [hunk 4]
```diff
@@ -131,3 +137,71 @@ export enum ApplicationFlags {
...
+	 */
+	DatetimeGreaterThanOrEqual,
+	/**
+	 * The metadata value (`integer`) is equal to the guild's configured value (`integer`; `1`)
+	 */
+	BooleanEqual,
+	/**
+	 * The metadata value (`integer`) is not equal to the guild's configured value (`integer`; `1`)
+	 */
+	BooleanNotEqual,
```
[the remark this answers is not in the document]
Comment (almeidx) [reply deno/payloads/v10/application.ts:206]: They match the ones on discords end
image
https://discord.com/developers/docs/resources/application-role-connection-metadata#application-role-connection-metadata-object-application-role-connection-metadata-type

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**